src/content/App/App.tsx

```diff
@@ -1,153 +1,156 @@
-    };
-
-  }, []); // Run only once on mount
-
-  // Update chrome storage whenever relevant state changes
-  useEffect(() => {
-    chrome.storage.sync.set({ sections });
-    // alert(JSON.stringify(sections, null, 2))
-  }, [sections]);
-
-  useEffect(() => {
-    // console.log(newSection)
-    if (newSection.code !== baseSection.code) {
-      if (newSection.term != Term.winterFull) {
-        //Don't set the term to WF, just keep the term to what is selected
-        setCurrentTerm(newSection.term);
-      }
-    }
-  }, [newSection]);
-
-  useEffect(() => {
-    chrome.storage.sync.set({ currentWorklistNumber });
-  }, [currentWorklistNumber]);
-
-  useEffect(() => {
-    chrome.storage.sync.set({ currentTerm });
-  }, [currentTerm]);
-
-  useEffect(() => {
-    chrome.storage.sync.set({ colorTheme });
-  }, [colorTheme]);
-
-  useEffect(() => {
-    // Check if there is a real change to trigger the update
-    // if (prevColorTheme.current !== colorTheme || JSON.stringify(prevSections.current) !== JSON.stringify(sections)) {
-    const newSections = assignColors(sections, colorTheme);
-
-    if (JSON.stringify(newSections) !== JSON.stringify(sections)) {
-      setSections(newSections);
-    }
-
-    // Update refs
-    // prevColorTheme.current = colorTheme;
-    // prevSections.current = sections;
-    // }
-  }, [colorTheme, sections]); // React only if these values change
-
-  return (
-    <div className="App">
-      <TopBar currentView={currentView} setCurrentView={setCurrentView} />
-      {currentView === Views.calendar ? (
-        <div className="CalendarViewContainer">
-          <CalendarContainer
-            sections={sections}
-            setSections={setSections}
-            newSection={newSection}
-            setInvalidSection={setInvalidSection}
-            currentWorklistNumber={currentWorklistNumber}
-            setCurrentWorklistNumber={setCurrentWorklistNumber}
-            currentTerm={currentTerm}
-            setCurrentTerm={setCurrentTerm}
-          />
-
-          <Form
-            currentWorklistNumber={currentWorklistNumber}
-            newSection={newSection}
-            sections={sections}
-            invalidSection={invalidSection}
-            setNewSection={setNewSection}
-            setSections={setSections}
-            currentTerm={currentTerm}
-            colorTheme={colorTheme}
-            setColorTheme={setColorTheme}
-          />
-        </div>
-      ) : (
-        <Settings 
-          colorTheme={colorTheme} 
-          sections={sections}
-          setColorTheme={setColorTheme}
-          setSections={setSections}
-        />
-      )}
-    </div>
-  );
+
+        // Update refs
+        // prevColorTheme.current = colorTheme;
+        // prevSections.current = sections;
+        // }
+    }, [colorTheme, sections]); // React only if these values change
+
+    return (
+        <HideProfilePictureProvider>
+            <div className="App">
```

Comment:
The goal was to hide the profile picture instantly when the page loads if the user has "Hide Profile Picture" checked. I've tried wrapping only Tools instead of the the entire App in the context, but that would only hide the profile picture when the calendar gets opened. 

I'll see if there's a better solution, though.

Reply:
I think I got it.

Reply:
I think I've got it.